File: lib/src/def.dart
/*
* Copyright 2019 mark , All right reserved
* author : mark
* date : 2019-11-05
* ide : VSCode
*/

/// 状态回调
const String DEF_CAMERA_STATUE_CALLBACK = "holo#livingStatueCallback";

/// 视图id
const String DEF_CAMERA_RTMP_VIEW = "holo#cameraRtmpView";

/// 配置方法
const String DEF_CAMERA_SETTING_CONFIG = "holo#cameraSettingConfig";

enum RtmpStatue {
  preparing,
  failed,
  living,
  pause,
  resume,
  stop,
}

abstract class MethodDef {
  static String get initConfig => "initConfig";
  static String get startLive => "startLive";
  static String get stopLive => "stopLive";
  static String get pauseLive => "pauseLive";
  static String get resumeLive => "resumeLive";
  static String get dispose => "dispose";
  static String get cameraRatio => "cameraRatio";
  static String get switchCamera => "switchCamera";
  static String get rotateOrientation => "rotateOrientation";
}

File: android/src/main/kotlin/holo/mark/flutter_rtmp/def.kt
package holo.mark.flutter_rtmp

// 状态回调
var DEF_CAMERA_STATUE_CALLBACK = "holo#livingStatueCallback"

/// 视图id
var DEF_CAMERA_RTMP_VIEW = "holo#cameraRtmpView"

/// 配置方法
var DEF_CAMERA_SETTING_CONFIG = "holo#cameraSettingConfig"


var PKG_NAME = "hv.rtmp"


// --------------  连接状态 --------------
/// 准备中
var RTMP_STATUE_Preparing = 0
/// 准备结束
var RTMP_STATUE_Ready = -1
/// 连接中
var RTMP_STATUE_Pending = 1
/// 已连接
var RTMP_STATUE_Pushing = 2
/// 已断开
var RTMP_STATUE_Stop = 3
/// 连接出错
var RTMP_STATUE_Error = 4
///  正在刷新
var RTMP_STATUE_Refresh = 5